lib/internal/modules/utils.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const {
+  ObjectCreate,
+  StringPrototypeEndsWith,
+} = primordials;
+const CJSLoader = require('internal/modules/cjs/loader');
+const { Module, readPackageScope } = CJSLoader;
+const { getOptionValue } = require('internal/options');
+
+
+function shouldUseESMLoader(filePath) {
+  /**
+   * @type {string[]} userLoaders A list of custom loaders registered by the user
+   * (or an empty list when none have been registered).
+   */
+  const userLoaders = getOptionValue('--experimental-loader');
+  /**
+   * @type {string[]} userImports A list of preloaded modules registered by the user
+   * (or an empty list when none have been registered).
+   */
+  const userImports = getOptionValue('--import');
+  if (userLoaders.length > 0 || userImports.length > 0)
+    return true;
+  // Determine the module format of the main
+  if (filePath && StringPrototypeEndsWith(filePath, '.mjs'))
+    return true;
+  if (!filePath || StringPrototypeEndsWith(filePath, '.cjs'))
+    return false;
+  const pkg = readPackageScope(filePath);
+  return pkg && pkg.data.type === 'module';
+}
+
+/**
+ * @param {string} filePath
+ * @returns {any}
+ * requireOrImport imports a module if the file is an ES module, otherwise it requires it.
+ */
+function requireOrImport(filePath) {
+  const useESMLoader = shouldUseESMLoader(filePath);
+  if (useESMLoader) {
+    const { esmLoader } = require('internal/process/esm_loader');
+    const { pathToFileURL } = require('internal/url');
+    const { isAbsolute } = require('path');
+    const file = isAbsolute(filePath) ? pathToFileURL(filePath).href : filePath;
+    return esmLoader.import(file, undefined, ObjectCreate(null));
```

Comment:
```suggestion
    return esmLoader.import(file, undefined, { __proto__: null });
```

Reply:
I wish we could have just 1 way to do this—I don't care which of the 3–4 I'm aware of.

Reply:
I don't understand why the proto approach - undeniable syntax that doesn't require a primordial or a function call - wouldn't be preferred, but either way I agree it'd be better for the linter to enforce a single wa